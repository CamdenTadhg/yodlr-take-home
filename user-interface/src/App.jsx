import React from 'react'
import {BrowserRouter, Routes, Route} from 'react-router-dom';
import Navbar from './Navbar';
import SignupForm from './SignupForm';
import Home from './Home';
import UserList from './UserList';
import EditUserForm from './EditUserForm';
import './App.css'

function App() {

  return (
    <div className='App'>
        <BrowserRouter>
          <Navbar />
          <div className='main'>
          <Routes>
            <Route path='/signup' element={<SignupForm/>}/>
            <Route path='/admin' element={<UserList/>}/>
            <Route path='/' element={<Home/>}/>
          </Routes>
          </div>
        </BrowserRouter>

    </div>
  )
}

export default App

//dynamic data on admin page
//admin creation of new accounts
//sign up form validation
//sorting/searching of users
//authentication/authorization
//optimize assets (minimize and/or bundle css/js)
//experiment with alternative designs (A/B Testing is important for registration!)
  //move all axios into a separate api file and use variable for URL
  //add redux
    //move delete and activate functions to store?
  //change variable names to be clearer
