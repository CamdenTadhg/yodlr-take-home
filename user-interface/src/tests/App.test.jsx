import {render} from '@testing-library/react';
import App from '../App';
import userEvent from '@testing-library/user-event';
import {it, expect} from 'vitest';

it('renders home page and navigates to signup page and back', async () => {
    const {getByText, getAllByText} = render(
        <App />
    );

    expect(getByText(/Welcome to Yodlr/i)).toBeInTheDocument();

    await userEvent.click(getByText(/Signup/i));

    expect(getByText(/Signup for Yodlr/i)).toBeInTheDocument;

    await userEvent.click(getByText(/Yodlr Home/i));

    expect(getByText(/Welcome to Yodlr/i)).toBeInTheDocument();
});


it('navigates to user list page', async () => {
    const {getByText} = render(
        <App />
    );

    await userEvent.click(getByText(/Admin/i));
    expect(getByText(/User List/i)).toBeInTheDocument;
    
});

it('renders without crashing', () => {
    render(
        <App/>
    );
});

it('matches snapshot', () => {
    const {asFragment} = render(
        <App/>)
    expect(asFragment()).toMatchSnapshot();
})
