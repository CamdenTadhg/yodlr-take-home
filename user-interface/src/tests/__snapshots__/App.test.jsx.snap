// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      id="Navbar"
    >
      <a
        href="/"
        id="Navbar-logo"
      >
        Yodlr Home
      </a>
      <a
        class="Navbar-navlink"
        href="/signup"
      >
        Signup
      </a>
      <a
        class="Navbar-navlink"
        href="/admin"
      >
        Admin
      </a>
    </div>
    <div
      id="UserList"
    >
      <form
        id="EditUserForm"
      >
        <h3
          id="EditUserForm-header"
        >
          Edit User Form
        </h3>
        <label
          for="email"
        >
          Email: 
        </label>
        <input
          id="email"
          name="email"
          type="text"
          value=""
        />
        <label
          for="firstName"
        >
          First Name: 
        </label>
        <input
          id="firstName"
          name="firstName"
          type="text"
          value=""
        />
        <label
          for="lastName"
        >
          Last Name: 
        </label>
        <input
          id="lastName"
          name="lastName"
          type="text"
          value=""
        />
        <button>
          Submit
        </button>
      </form>
      <h1
        id="UserList-header"
      >
        User List
      </h1>
      <table
        id="UserList-table"
      >
        <thead>
          <tr>
            <th>
              Email
            </th>
            <th>
              First Name
            </th>
            <th>
              Last Name
            </th>
            <th />
            <th />
            <th />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</DocumentFragment>
`;
